cmd/meroxa/root/resources/rotate_tunnel_key.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2021 Meroxa Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"errors"
+
+	"github.com/meroxa/cli/cmd/meroxa/builder"
+
+	"github.com/meroxa/cli/log"
+
+	"github.com/meroxa/meroxa-go"
+)
+
+type rotateKeyActionClient interface {
+	RotateTunnelKeyForResource(ctx context.Context, name string) (*meroxa.Resource, error)
+}
+
+type RotateTunnelKey struct {
+	client rotateKeyActionClient
+	logger log.Logger
+
+	args struct {
+		Name string
+	}
+
+	flags struct {
+	}
+}
+
+func (u *RotateTunnelKey) Usage() string {
+	return "rotate-tunnel-key NAME"
+}
+
+func (u *RotateTunnelKey) Docs() builder.Docs {
+	return builder.Docs{
+		Short: "Rotate the tunnel key for a resource",
+		Long:  "Rotate the tunnel key for a Meroxa resource.",
+	}
+}
+
+func (u *RotateTunnelKey) Confirm(ctx context.Context) string {
+	u.logger.Infof(ctx, "Rotating tunnel key will restart the tunnel and disconnect existing connections.")
+	return u.args.Name
+}
+
+func (u *RotateTunnelKey) Execute(ctx context.Context) error {
+	r, err := u.client.RotateTunnelKeyForResource(ctx, u.args.Name)
+	if err != nil {
+		return err
+	}
+
+	u.logger.Infof(ctx, "Resource %q tunnel key is successfully rotated!", u.args.Name)
+	if tun := r.SSHTunnel; tun != nil {
+		u.logger.Info(ctx, "Paste the following public key on your host:")
+		u.logger.Info(ctx, tun.PublicKey)
+		u.logger.Info(ctx, "Meroxa will try to connect to the resource for 60 minutes and send an email confirmation after a successful resource validation.") //nolint
+	}
+	u.logger.JSON(ctx, r)
+
+	return nil
+}
+
+func (u *RotateTunnelKey) Flags() []builder.Flag {
+	return builder.BuildFlags(&u.flags)
+}
+
+func (u *RotateTunnelKey) Logger(logger log.Logger) {
+	u.logger = logger
+}
+
+func (u *RotateTunnelKey) Client(client *meroxa.Client) {
+	u.client = client
+}
+
+func (u *RotateTunnelKey) ParseArgs(args []string) error {
+	if len(args) < 1 {
+		return errors.New("requires resource name")
+	}
+
+	u.args.Name = args[0]
+	return nil
+}
+
+var (
```

Comment:
If I remember correctly, @raulb wants to keep these vars near the top of the file. Think: https://github.com/meroxa/cli/blob/ae8f69c52aa2fcd581f12562b74b65c1619f2c0e/cmd/meroxa/root/resources/list.go#L28

Reply:
Sure, i can do that.